### PR TITLE
Add Unit test for staking/types

### PR DESCRIPTION
**Pull request by Renewwen**
Test functions:
```
NewDelegation()
Undelegation()
TotalInUndelegation()
DeleteEntry()
RemoveUnlockedUndelegations()
```

### CODE REVIEW

**Comment by harmony-ek** [comment]
Jenkins speaking; can one of the admins verify this patch?

**Comment by Renewwen** [comment]
### Add Unit test for `validator.go`
> Test coverage: 18.3% ---> 44.7%

image

**Comment by Daniel-VDM** [comment]
@Renewwen Fix Travis pls

**Comment by LeoHChen** [comment]
Jenkins, test this please.

File: staking/types/validator_test.go [hunk 1]
```diff
@@ -1,18 +1,248 @@
...
+	return ValidatorWrapper{
+		Validator: v,
+	}
+}
+
+// Test Validator's MarshalJSON
+func TestMarshalJSON(t *testing.T) {
+	_, err := validator.MarshalJSON()
+	if err != nil {
+		t.Errorf("change Validator to Marsha JSON failed")
```
Comment (gupadhyaya) [inline staking/types/validator_test.go]: fix the typos and make the error messages more meaningful.

**Comment by gupadhyaya** [comment]
```
PASS
ok  	github.com/harmony-one/harmony/staking/types	0.068s	coverage: 51.3% of statements
```

**Comment by LeoHChen** [comment]
Jenkins, test this please.

**Comment by fxfactorial** [comment]
Jenkins, test this please

**Pull request merged**